p2p/libp2p/netMessenger.go

```diff
@@ -461,10 +461,15 @@ func (netMes *networkMessenger) RegisterMessageProcessor(topic string, handler p
 		netMes.Broadcast(topic, buffToSend)
 	}
 
-	err := netMes.pb.RegisterTopicValidator(topic, func(ctx context.Context, pid peer.ID, message *pubsub.Message) bool {
-		err := handler.ProcessReceivedMessage(NewMessage(message), broadcastHandler)
+	err := netMes.pb.RegisterTopicValidator(topic, func(ctx context.Context, pid peer.ID, pubsubMessage *pubsub.Message) bool {
+		message, err := NewMessage(pubsubMessage)
 		if err != nil {
-			log.Trace("p2p validator", "error", err.Error(), "topics", message.TopicIDs)
+			log.Trace("p2p validator - new message", "error", err.Error(), "topics", pubsubMessage.TopicIDs)
+			return false
+		}
+		err = handler.ProcessReceivedMessage(message, broadcastHandler)
+		if err != nil {
+			log.Trace("p2p validator", "error", err.Error(), "topics", pubsubMessage.TopicIDs)
```

Comment:
I would say that is better to return directly false here and at the end of the method to simply return true, because if somebody will extend this method after this point, could forgot to add a return false here in the case of an error

Reply:
done